pytorch_lightning/metrics/functional/precision_recall.py

```diff
@@ -0,0 +1,496 @@
+            than what they appear to be. See the parameter's
+            :ref:`documentation section <metrics:Using the is_multiclass parameter>`
+            for a more detailed explanation and examples.
+
+        class_reduction:
+            .. warning :: This parameter is deprecated, use ``average``. Will be removed in v1.4.0.
+
+    Return:
+        The shape of the returned tensor depends on the ``average`` parameter
+
+        - If ``average in ['micro', 'macro', 'weighted', 'samples']``, a ``(2, )`` tensor will be returned
+        - If ``average in ['none', None]``, the shape will be ``(2, C)``, where ``C`` stands  for the number
+          of classes
+
+        The first element (in the first dimension) corresponds to precision, the second one to recall.
+
+    Example:
+
+        >>> from pytorch_lightning.metrics.functional import precision_recall
+        >>> preds  = torch.tensor([2, 0, 2, 1])
+        >>> target = torch.tensor([1, 1, 2, 0])
+        >>> precision_recall(preds, target, average='macro', num_classes=3)
+        tensor([0.1667, 0.3333])
+        >>> precision_recall(preds, target, average='micro')
+        tensor([0.2500, 0.2500])
+
+    """
+    if class_reduction:
+        rank_zero_warn(
+            "This `class_reduction` parameter was deprecated in v1.2.0 in favor of"
+            " `reduce`. It will be removed in v1.4.0",
+            DeprecationWarning,
+        )
+        average = class_reduction
+
+    allowed_average = ["micro", "macro", "weighted", "samples", "none", None]
+    if average not in allowed_average:
+        raise ValueError(f"The `average` has to be one of {allowed_average}, got {average}.")
+
+    allowed_mdmc_average = [None, "samplewise", "global"]
+    if mdmc_average not in allowed_mdmc_average:
+        raise ValueError("The `mdmc_average` has to be one of {allowed_mdmc_average}, got {mdmc_average}.")
+
+    if average in ["macro", "weighted", "none", None] and (not num_classes or num_classes < 1):
+        raise ValueError(f"When you set `average` as {average}, you have to provide the number of classes.")
+
+    if num_classes and ignore_index is not None and (not 0 <= ignore_index < num_classes or num_classes == 1):
+        raise ValueError(f"The `ignore_index` {ignore_index} is not valid for inputs with {num_classes} classes")
+
+    reduce = "macro" if average in ["weighted", "none", None] else average
+    tp, fp, tn, fn = _stat_scores_update(
+        preds,
+        target,
+        reduce=reduce,
+        mdmc_reduce=mdmc_average,
+        threshold=threshold,
+        num_classes=num_classes,
+        top_k=top_k,
+        is_multiclass=is_multiclass,
+        ignore_index=ignore_index,
+    )
+
+    precision = _precision_compute(tp, fp, tn, fn, average, mdmc_average)
+    recall = _recall_compute(tp, fp, tn, fn, average, mdmc_average)
+
+    return torch.stack([precision, recall])
```

Comment:
I would prefer if this function returned a tuple of scores instead of stacking them together

Reply:
No problem